SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -375,32 +395,59 @@ void HGCDigitizer::accumulate(edm::Handle<edm::PCaloHitContainer> const &hits,
 
     (simHitIt->second).hit_info[0][itime] += charge;
     float accCharge=(simHitIt->second).hit_info[0][itime];
-      
+
+
+    //working version with pileup only for in-time hits
+    int waferThickness = getCellThickness(geom,id);
+    float accChargeForToA = 0.f;
+    bool orderChanged = false;
+    if(itime == 9){
+      if(hitRefs_bx0[id].empty()){
+	hitRefs_bx0[id].push_back(std::pair<float, float>(charge, tof));
+	accChargeForToA += charge;
+      }
+      else if(tof <= hitRefs_bx0[id].back().second){
+	hitRefs_bx0[id].push_back(std::pair<float, float>(charge, tof));
+	std::sort(hitRefs_bx0[id].begin(), hitRefs_bx0[id].end(), comparePairs);
+	for(const auto& step : hitRefs_bx0[id]){
+	  accChargeForToA += step.first;
+	  if(accChargeForToA > tdcForToaOnset[waferThickness-1] && step.second != hitRefs_bx0[id].back().second){
+	    while(step != hitRefs_bx0[id].back())  hitRefs_bx0[id].pop_back();
```

Comment:
I'm not entirely sure I understand what this is doing, but maybe it could be more efficient to use a binary search (e.g. `std::lower_bound`), since the vector is sorted?